src/client.ts

```diff
@@ -85,15 +86,41 @@ export default class UnleashClient extends EventEmitter {
           this.warnOnce(strategySelector.name, feature.name, feature.strategies);
           return false;
         }
-        return strategy.isEnabledWithConstraints(
-          strategySelector.parameters,
-          context,
-          strategySelector.constraints,
-        );
+        const constraints = this.yieldConstraintsFor(strategySelector);
```

Comment:
this is quite elegant !

Reply:
Thank you! I'm a fan as well, generators make this really easy to do